src/anndata/_io/specs/lazy_methods.py

```diff
@@ -36,7 +36,7 @@
 @contextmanager
 def maybe_open_h5(
     path_or_group: Path | ZarrGroup, elem_name: str
-) -> Generator[StorageType, None, None]:
+) -> Callable[[], Iterator[StorageType]]:
```

Comment:
The previous type was correct, the new one isn’t.

The way decorators interact with Python is that you normally type the decorated function (e.g. if it contains `yield`, the return type is `Generator`). The decorator than transforms the function from whatever it is to whatever the decorator wants.